specification/configuration.md

```diff
@@ -89,6 +90,50 @@ environment variables:
   If `SPLUNK_CONFIG` is defined then these options are not required but MAY be
   defined.
 
+### Kubernetes
+
+While Kubernetes supports container technology that can be configured using
+environment variables, package management solutions such as Helm charts and
+Operators require YAML-based configuration. As a result, Kubernetes package
+management solutions MUST support the following YAML configuration options:
+
+- `splunkCommon`
```

Comment:
Assuming we target helm initially in this specification. The existing configuration interface in the helm chart tend to be flat and don't use grouping unless it's necessary following helm best practices https://helm.sh/docs/chart_best_practices/values/#flat-or-nested-values

Do we need this particular grouping under `splunkCommon`? Can we assume any flat configuration other than `splunk<Observability|Platform>` considered to be common?

Reply:
The guidance in the link appears to go both way. For example: `When there are a large number of related variables, and at least one of them is non-optional, nested values may be used to improve readability.` My concern is that there are a lot of additional options and lack of nesting for confusing for me.

What if we do:

```
splunk:
  environment:
  observability:
    accessToken:
  platform:
    token:
```

Reply:
+1 this is what the operator does as well i.e, it accepts "common" fields at the root level and then has nested levels for different components e.g, 

```yaml
spec:
  clusterName: <MY_CLUSTER_NAME>
  realm: <SPLUNK_REALM>
  agent:
     ...
  gateway:
     ...
```

https://github.com/signalfx/splunk-otel-collector-operator/blob/main/docs/splunkotelagent_cr_spec.md